src/compose/service.ts

```diff
@@ -684,17 +689,31 @@ export class Service {
 		let sameConfig = _.isEqual(thisOmitted, otherOmitted);
 		const nonArrayEquals = sameConfig;
 
-		// Check for array fields which don't match
+		// Check for array fields which don't match. We first consider
+		// unordered fields, and then ordered
 		const differentArrayFields: string[] = [];
 		sameConfig =
 			sameConfig &&
 			_.every(Service.configArrayFields, (field: keyof ServiceConfig) => {
-				const eq = _.isEqual(this.config[field], service.config[field]);
+				const eq = _.isEqual(
+					_.sortBy(this.config[field] as unknown[]),
+					_.sortBy(service.config[field] as unknown[]),
+				);
 				if (!eq) {
 					differentArrayFields.push(field);
 				}
 				return eq;
-			});
+			}) &&
+			_.every(
+				Service.orderedConfigArrayFields,
+				(field: keyof ServiceConfig) => {
+					const eq = _.isEqual(this.config[field], service.config[field]);
+					if (!eq) {
+						differentArrayFields.push(field);
```

Comment:
This will never be reached if the previous `_.every(...)` return false, making `differentArrayFields` potentially incorrect (missing some ordered fields).